be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+                    std::cout << col->get_element(i) << " ";
+                }
+            } else if (WhichDataType(dataType).is_decimal256()) {
+                auto col = check_and_get_column<ColumnDecimal<Decimal256>>(column);
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << col->get_element(i) << " ";
+                }
+            } else if (WhichDataType(dataType).is_date()) {
+                auto col = check_and_get_column<ColumnDate>(column);
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << col->get_data_at(i) << " ";
+                }
+            } else if (WhichDataType(dataType).is_date_time()) {
+                auto col = check_and_get_column<ColumnDateTime>(column);
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << col->get_data_at(i) << " ";
+                }
+            } else if (WhichDataType(dataType).is_date_v2()) {
+                auto col = check_and_get_column<ColumnDateV2>(column);
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << col->get_data_at(i) << " ";
+                }
+            } else if (WhichDataType(dataType).is_date_time_v2()) {
+                auto col = check_and_get_column<ColumnDateTimeV2>(column);
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << col->get_data_at(i) << " ";
+                }
+            } else {
+                std::cout << "data type: " << dataType.get_name() << std::endl;
+                std::cout << "column type: " << column.get_name() << std::endl;
+                for (size_t i = 0; i < column_size; ++i) {
+                    std::cout << column.get_data_at(i).to_string() << " ";
+                }
+            }
+            std::cout << std::endl;
+        }
+    }
+    // column size changed calculation:
+    //  size, reserve, resize, empty, byte_size, allocated_bytes, clone_resized, get_shrinked_column
+    //  cut(LIMIT operation), shrink
+    void sizeAssert(MutableColumnPtr col, size_t expect_size) {
+        EXPECT_EQ(col->size(), expect_size);
+    }
+
+    // empty just use size() == 0 to impl as default behavior
+    void emptyAssert(MutableColumnPtr col) { EXPECT_EQ(col->size(), 0); }
+
+    // reserve, resize, byte_size, allocated_bytes, clone_resized, get_shrinked_column
+    void reserveAssert(MutableColumnPtr col, size_t expect_size) {
+        col->reserve(expect_size);
+        EXPECT_EQ(col->allocated_bytes(), expect_size);
+    }
+
+    //  cut(LIMIT operation) will cut the column with the given from and to, and return the new column
+    //  notice return column is clone from origin column
+    void cutAssert(MutableColumnPtr col, size_t from, size_t to, size_t expect_size) {
+        auto ori = col->size();
+        auto ptr = col->cut(from, to);
+        EXPECT_EQ(ptr->size(), expect_size);
```

Comment:
ommit expect_size argument and use to - from

Reply:
done